ffi/examples/read-table/read_table.c

```diff
@@ -79,6 +79,18 @@ EngineError* allocate_error(KernelError etype, const KernelStringSlice msg)
   return (EngineError*)error;
 }
 
+#ifdef WIN32 // windows doesn't have strndup
+char *strndup(const char *s, size_t n) {
+  size_t len = strnlen(s, n);
```

Comment:
does this blow up if s is `NULL`? should check if it's null and just return null if so?

Reply:
Yes, but this shouldn't be called on nulls to start, a lot of these older windows functions don't check for null (or they have null safe variants)

Reply:
Yep, but so does strndup (it'll segfault), so this is pretty much equivalent. (yay c)